src/components/crowdsale/utils.js

```diff
@@ -182,42 +182,69 @@ export let getCrowdsaleTargetDates = (initCrowdsaleContract, execID) => {
 
     let registryStorageObj = toJS(contractStore.registryStorage)
 
-    let getCrowdsaleStartAndEndTimes = initCrowdsaleContract.methods.getCrowdsaleStartAndEndTimes(registryStorageObj.addr, execID).call();
-    let getCurrentTierInfo = crowdsaleStore.isMintedCappedCrowdsale ? initCrowdsaleContract.methods.getCurrentTierInfo(registryStorageObj.addr, execID).call() : null;
+    if (crowdsaleStore.isMintedCappedCrowdsale) {
+      return getTiersLength()
+        .then((tiersLength) => {
+          let getTiersStartAndEndDates = []
+          for (let ind = 0; ind < tiersLength; ind++) {
+            let getTierStartAndEndDates = initCrowdsaleContract.methods.getTierStartAndEndDates(registryStorageObj.addr, execID, ind).call()
+            getTiersStartAndEndDates.push(getTierStartAndEndDates)
+          }
+
+          return Promise.all(getTiersStartAndEndDates)
+        })
+        .then((tiersStartAndEndDates) => {
+          console.log("tiersStartAndEndDates:", tiersStartAndEndDates)
+          let crowdsaleStartDate
+          let crowdsaleEndDate = 0
+          tiersStartAndEndDates.forEach((tierStartAndEndDates) => {
+            const tierDates = setTierDates(tierStartAndEndDates.tier_start, tierStartAndEndDates.tier_end)
+            crowdsaleStartDate = crowdsaleStartDate ? Math.min(crowdsaleStartDate, tierDates.startsAtMilliseconds) : tierDates.startsAtMilliseconds
```

Comment:
Is it necessary to make those comparisons (`min`, `max`)?

If you're requesting tiers in order, `tiersStartAndEndDates` must be ordered as well. Thus, just picking the `tier_start` from the first tier, and `tier_end` from the last one should be enough?